lib/python/kle2xy.py

```diff
@@ -113,16 +100,9 @@ def parse_layout(self, layout):  # noqa  FIXME(skullydazed): flake8 says this ha
                     current_key['row'] = round(current_row, 2)
                     current_key['column'] = round(current_col, 2)
 
-                    # Determine the X center
-                    x_center = (current_key['width'] * self.key_width) / 2
-                    current_x += x_center
-                    current_key['x'] = current_x
-                    current_x += x_center
-
-                    # Determine the Y center
-                    y_center = (current_key['height'] * self.key_width) / 2
-                    y_offset = y_center - (self.key_width / 2)
-                    current_key['y'] = (current_y + y_offset)
+                    # x,y (units mm) is the center of the key
+                    current_key['x'] = (current_col + current_key['width'] / 2) * self.key_width
+                    current_key['y'] = (current_row + current_key['height'] / 2) * self.key_width
```

Comment:
This is a good optimization, but this now has several places that can throw an exception in one line. Better to break it up a bit for readability and debugability.

```suggestion
                    x_center = current_col + (current_key['width'] / 2)
                    y_center = current_row + (current_key['height'] / 2)
                    current_key['x'] = x_center * self.key_width
                    current_key['y'] = y_center * self.key_width
```